src/hooks/useInterval.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useRef } from "react";
+
+/**
+ * React hook that simplifies using setInterval() within functional components.
+ *
+ * Note: You don't need to memoise/useCallback the handler as we store it with
+ * a ref internally.
+ *
+ * @param handler the function to be called every tick of the interval
+ * @param intervalInMs the interval time in milliseconds
+ */
+export const useInterval = (handler: () => void, intervalInMs?: number) => {
```

Comment:
We should add an explicit callout about what happens when `intervalInMs` is `undefined`, and maybe some `@example` about how to use this hook in both the normal case, and the 'conditionally run' case, as called out in https://github.com/sparkletown/sparkle/pull/890#discussion_r536484984

Reply:
Yep agreed; from what I can see the case where `intervalMs` is `undefined` does nothing, so we could shortcut out? Just trying to understand the intent behind making `intervalMs` optional.

It seems equivalent to:

```
export const useInterval = (handler: () => void, intervalInMs?: number) => {
  if (!intervalMs) return;
  ...
```

Reply:
You can't conditionally render hooks. If you tried to bail out there, react would complain, as you have branching logic before the `useEffect`'s/etc below.

If you look at https://github.com/sparkletown/sparkle/pull/890/files/82625de2d8cfa326c4b2127dfdc6fe28fd086fbc#diff-1731fae0e569c8786cc66cc2d47102f32a35838a19aa9d417647cab9db39cacdR21-R31 you can see that we have the early bailout in the `useEffect`, so that the interval will never be created.

So all that needs to change here is adding some extra documentation/explanation in the JSdocs to make this obvious to people.

Reply:
Yep, just needs to be documented.

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/890/commits/2c8050bf8c7a11cd615de82f3f800358a5f41a77